src/test/java/com/uber/cadence/workflow/LoggerTest.java

```diff
@@ -0,0 +1,136 @@
+import java.util.List;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.slf4j.Logger;
+
+@RunWith(Parameterized.class)
+public class LoggerTest {
+  private static final String taskList = "logger-test";
+  private static final Logger workflowLogger = Workflow.getLogger(TestLoggingInWorkflow.class);
+  private static final Logger childWorkflowLogger =
+      Workflow.getLogger(TestLoggerInChildWorkflow.class);
+
+  @Parameters
+  public static Collection<Object[]> data() {
+    return Arrays.asList(new Object[][] {{false, "wf-1", 1}, {true, "wf-2", 3}});
+  }
+
+  @Parameter public boolean loggingEnabledInReplay;
+
+  @Parameter(1)
+  public String wfID;
+
+  @Parameter(2)
+  public int startLinesExpected;
+
+  public interface TestWorkflow {
+    @WorkflowMethod
+    void execute(String id);
+  }
+
+  public static class TestLoggingInWorkflow implements LoggerTest.TestWorkflow {
+    @Override
+    public void execute(String id) {
+      workflowLogger.info("Start executing workflow {}.", id);
+      ChildWorkflowOptions options =
+          new ChildWorkflowOptions.Builder().setTaskList(taskList).build();
+      LoggerTest.TestChildWorkflow workflow =
+          Workflow.newChildWorkflowStub(LoggerTest.TestChildWorkflow.class, options);
+      workflow.executeChild(id);
+      workflowLogger.info("Done executing workflow {}.", id);
+    }
+  }
+
+  public interface TestChildWorkflow {
+    @WorkflowMethod
+    void executeChild(String id);
+  }
+
+  public static class TestLoggerInChildWorkflow implements LoggerTest.TestChildWorkflow {
+
+    @Override
+    public void executeChild(String id) {
+      childWorkflowLogger.info("Executing child workflow {}.", id);
+    }
+  }
+
+  @Test
+  public void testWorkflowLogger() throws IOException {
+    TestEnvironmentOptions testOptions =
+        new Builder()
+            .setDomain(WorkflowTest.DOMAIN)
+            .setEnableLoggingInReplay(loggingEnabledInReplay)
+            .build();
+    TestWorkflowEnvironment env = TestWorkflowEnvironment.newInstance(testOptions);
+    Worker worker = env.newWorker(taskList);
+    worker.registerWorkflowImplementationTypes(
+        TestLoggingInWorkflow.class, TestLoggerInChildWorkflow.class);
+    worker.start();
+
+    WorkflowClient workflowClient = env.newWorkflowClient();
+    WorkflowOptions options =
+        new WorkflowOptions.Builder()
+            .setExecutionStartToCloseTimeout(Duration.ofSeconds(1000))
+            .setTaskList(taskList)
+            .build();
+    LoggerTest.TestWorkflow workflow =
+        workflowClient.newWorkflowStub(LoggerTest.TestWorkflow.class, options);
+    workflow.execute(wfID);
+
+    String tempDir = System.getProperty("java.io.tmpdir");
+    System.out.println(tempDir);
+    File logFile = new File(tempDir, "log-cadence-java-client.log");
```

Comment:
I think you can avoid all this complexity of using a file and configuring it. 
Use http://oct.im/how-to-create-logback-loggers-dynamicallypragmatically.html
with https://github.com/qos-ch/logback/blob/master/logback-core/src/main/java/ch/qos/logback/core/read/ListAppender.java